### DISCUSSION TITLE
Is there a way to define models versions differently in the model repository?

### DISCUSSION THREAD

**Discussion by RafayAK** [triton-inference-server/server#5411] [Q&A] [answered]
Hi, I've been testing the basics of the triton inference server and would like to know if there is a way to define model versions differently than the whole number system currently (1, 2, 3, 4, ...). 

According to the docs we need to define the `model-repository` directory as follows:
```
model_repository
├── my-detection-model
│   ├── 1
│   │   └── model.pt
|   ├── 2
│   │   └── model.pt
|   ├── 3
│   │   └── model.pt
│   └── config.pbtxt
```

What I would prefer is to do is somewhat follow the [semantic versioning policy ](https://semver.org/), something like this:

```
model_repository
├── my-detection-model
│   ├── 1.0
│   │   └── model.pt
|   ├── 1.1
│   │   └── model.pt
|   ├── 1.2
│   │   └── model.pt
│   └── config.pbtxt
```

**Comment by the-david-oy** [comment, marked as answer]
I don't believe so. Throughout the code, the model version is [stored as an integer](https://github.com/triton-inference-server/core/blob/b6070e284cb37d90cba1ad5526712fbf06748b0d/src/infer_trace.h#L63), so the version would need to be an integer.

**Comment by RafayAK** [comment]
@dyastremsky Thanks for clearing that up! Couldn't find clear documentation if such a thing was supported/possible.
